gsuite/resource_group.go

```diff
@@ -75,6 +77,10 @@ func resourceGroupCreate(d *schema.ResourceData, meta interface{}) error {
 
 	createdGroup, err := config.directory.Groups.Insert(group).Do()
 	if err != nil {
+		if strings.Contains(err.Error(), "quotaExceeded") {
```

Comment:
Can you create a function and send the callback as a last argument instead? Not sure that works in go, but repeating the same code block over and over doesn't seem very elegant.

Reply:
You can use [this](https://github.com/terraform-providers/terraform-provider-google/blob/aaac3a1df95f65f701201af7d4477c6bd394b2cd/google/resource_storage_bucket.go#L300) as an example. Or [this](https://github.com/terraform-providers/terraform-provider-google/blob/aaac3a1df95f65f701201af7d4477c6bd394b2cd/google/resource_storage_bucket.go#L480) for a more Terraformish way.

Reply:
@drebes @psalaberria002 thank you both for the feedback, I updated the PR 😄 

@psalaberria002 I'll test your PR soon, do you think you can base your changes on the `quotas` branch and als use this same functions?

Reply:
Sure, no problem. I need to look into pagination also.